platform/tezos/model.go

```diff
@@ -1,94 +1,114 @@
+	"time"
+)
 
 const (
-	TxTransactions TxType = "transactions"
-	TxDelegations  TxType = "delegations"
+	TxTypeTransaction string = "transaction"
+	TxTypeDelegation  string = "delegation"
 
-	TxKindTransaction TxKind = "transaction"
-	TxKindDelegation  TxKind = "delegation"
-
-	TxStatusApplied TxStatus = "applied"
-	TxStatusSkipped TxStatus = "skipped"
+	TxStatusApplied string = "applied"
 )
 
-type Op struct {
-	OpHash         string    `json:"opHash"`
-	BlockLevel     uint64    `json:"blockLevel"`
-	BlockTimestamp time.Time `json:"blockTimestamp"`
+type Account struct {
+	Balance  string `json:"balance"`
+	Delegate string `json:"delegate"`
+}
+
+type ExplorerAccount struct {
+	Transactions []Transaction `json:"ops"`
 }
 
 type Transaction struct {
-	Tx         Tx         `json:"tx"`
-	Op         Op         `json:"op"`
-	Delegation Delegation `json:"delegation"`
+	Delegate  string  `json:"delegate"` // Current delegate (may be self when registered as delegate).
+	Errors    []Error `json:"errors"`   // Operation status applied, failed, backtracked, skipped.
+	Fee       float64 `json:"fee"`      // Total fee paid (and frozen) by all operations.
+	Hash      string  `json:"hash"`     // Operation hash.
+	Height    uint64  `json:"height"`
+	IsSuccess bool    `json:"is_success"` // Flag indicating operation was successfully applied.
+	Receiver  string  `json:"receiver"`
+	Sender    string  `json:"sender"`
+	Stat      string  `json:"status"` // Operation status applied, failed, backtracked, skipped.
+	Time      string  `json:"time"`   // Block time at which the operation was included on-chain e.g: 2019-09-28T13:10:51Z
+	Type      string  `json:"type"`   // Operation type, one of activate_account, double_baking_evidence, double_endorsement_evidence, seed_nonce_revelation, transaction, origination, delegation, reveal, endorsement, proposals, ballot.
+	Volume    float64 `json:"volume"`
 }
 
-func (t *Transaction) Status() TxStatus {
-	if len(t.Tx.Status) > 0 {
-		return t.Tx.Status
-	}
-	return t.Delegation.Status
+type Error struct {
+	ID   string `json:"id"`
+	Kind string `json:"kind"`
 }
 
-func (t *Transaction) Kind() TxKind {
-	if len(t.Tx.Kind) > 0 {
-		return t.Tx.Kind
-	}
-	return t.Delegation.Kind
+type Status struct {
+	Indexed int64 `json:"indexed"`
 }
 
-func (t *Transaction) Source() string {
-	if len(t.Tx.Source) > 0 {
-		return t.Tx.Source
-	}
-	return t.Delegation.Source
+type Validator struct {
+	Address string `json:"pkh"`
 }
 
-func (t *Transaction) Destination() string {
-	if len(t.Tx.Destination) > 0 {
-		return t.Tx.Destination
+func (t *Transaction) Status() blockatlas.Status {
+	switch t.Stat {
+	case TxStatusApplied:
+		return blockatlas.StatusCompleted
+	default:
+		return blockatlas.StatusError
 	}
-	return t.Delegation.Delegate
 }
 
-func (t *Transaction) Fee() string {
-	if len(t.Tx.Fee) > 0 {
-		return t.Tx.Fee
+func (t *Transaction) ErrorMsg() string {
+	switch t.IsSuccess {
```

Comment:
Maybe it’s better not to use a switch here?

Reply:
https://github.com/trustwallet/blockatlas/pull/980/commits/64671a7ef3e517b2e52cd479ab40b70b1e56a565